src/System.Buffers.Experimental/ReferenceCountedBuffer.cs

```diff
@@ -17,15 +17,16 @@ public override void Retain()
             Interlocked.Increment(ref _referenceCount);
         }
 
-        public override void Release()
+        public override bool Release()
         {
             if (!IsRetained) BuffersExperimentalThrowHelper.ThrowInvalidOperationException();
             if (Interlocked.Decrement(ref _referenceCount) <= 0) {
                 OnNoReferences();
             }
+            return IsRetained;
         }
 
-        public override bool IsRetained => _referenceCount > 0;
+        protected override bool IsRetained => _referenceCount > 0;
```

Comment:
OK, I think I know what you mean. I was confused cause your comment is below the IsRetained property. Yes, we need to make the implementation of Release better.